src/dotty/tools/dotc/transform/GetClass.scala

```diff
@@ -0,0 +1,45 @@
+package dotty.tools.dotc
+package transform
+
+import ast.tpd
+import core.Constants.Constant
+import core.Contexts.Context
+import core.StdNames.nme
+import core.Symbols.TermSymbol
+import core.TypeErasure
+import TreeTransforms.{MiniPhaseTransform, TransformerInfo}
+
+/**
+ * FILL ME
+ */
+class GetClass extends MiniPhaseTransform {
+  import tpd.{Apply, Tree, Literal, ref, TreeOps}
+
+  override def phaseName: String = "getClass"
+
+  override def transformApply(tree: Apply)(implicit ctx: Context, info: TransformerInfo): Tree = {
+    import ast.Trees.{Apply, Select}
+
+    tree match {
+      case Apply(Select(typ, nm), List()) if nm.toString == "getClass" =>
```

Comment:
I would rename `typ` to `qual` and `nm` to `name`.
Also, instead of `nm.toString == "getClass"` you should do `nm == nme.getClass`